core/templates/components/question-directives/question-misconception-editor/question-misconception-editor.component.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the question misconception editor.
+ */
+
+require('directives/angular-html-bind.directive.ts');
+
+require('domain/utilities/url-interpolation.service.ts');
+require(
+  'components/state-editor/state-editor-properties-services/' +
+  'state-editor.service.ts');
+require(
+  'components/question-directives/question-misconception-selector/' +
+  'question-misconception-selector.component.ts');
+require(
+  'components/question-directives/question-misconception-editor/' +
+  'tag-misconception-modal.controller.ts');
+
+angular.module('oppia').component('questionMisconceptionEditor', {
+  bindings: {
+    getOnSaveAnswerGroupFeedbackFn: '&onSaveAnswerGroupFeedback',
+    getOnSaveTaggedMisconception: '&onSaveTaggedMisconception',
+    getTaggedSkillMisconceptionId: '&taggedSkillMisconceptionId',
+    isEditable: '=',
+    outcome: '=',
+    rules: '=',
+  },
+  template: require('./question-misconception-editor.component.html'),
+  controller: [
+    '$rootScope', '$uibModal', 'StateEditorService',
+    'UrlInterpolationService',
+    function(
+        $rootScope, $uibModal, StateEditorService,
+        UrlInterpolationService) {
+      var ctrl = this;
+      var _getTaggedMisconceptionName = function(skillMisconceptionId) {
```

Comment:
This isn't really a getter since it's not returning anything. Can you pick a better name?

I'm also somewhat confused by why this is invoked in [line 114](https://github.com/oppia/oppia/pull/9337/files#diff-66e013a640f0258dfea0816c32c697eeR114) -- you seem to be combining the pieces, only to split them again in this function.